central/reports/scheduler/v2/reportgenerator/report_gen_view_based_test.go

```diff
@@ -0,0 +1,638 @@
+				componentNames:  []string{"c1_ns1_dep0_img_comp", "c1_ns2_dep0_img_comp", "c2_ns1_dep0_img_comp", "c2_ns2_dep0_img_comp", "w0_img_comp", "w1_img_comp"},
+				cveNames: []string{
+					"CVE-fixable_critical-c1_ns1_dep0_img_comp",
+					"CVE-fixable_critical-c1_ns2_dep0_img_comp",
+					"CVE-fixable_critical-c2_ns1_dep0_img_comp",
+					"CVE-fixable_critical-c2_ns2_dep0_img_comp",
+					"CVE-fixable_critical-w0_img_comp",
+					"CVE-fixable_critical-w1_img_comp",
+				},
+			},
+		},
+		{
+			name:  "View-based report with empty query (should return all vulnerabilities)",
+			query: "",
+			expected: &viewBasedReportData{
+				deploymentNames: []string{"c1_ns1_dep0", "c1_ns2_dep0", "c2_ns1_dep0", "c2_ns2_dep0"},
+				imageNames:      []string{"c1_ns1_dep0_img", "c1_ns2_dep0_img", "c2_ns1_dep0_img", "c2_ns2_dep0_img", "w0_img", "w1_img"},
+				componentNames:  []string{"c1_ns1_dep0_img_comp", "c1_ns2_dep0_img_comp", "c2_ns1_dep0_img_comp", "c2_ns2_dep0_img_comp", "w0_img_comp", "w1_img_comp"},
+				cveNames: []string{
+					"CVE-fixable_critical-c1_ns1_dep0_img_comp", "CVE-nonFixable_low-c1_ns1_dep0_img_comp",
+					"CVE-fixable_critical-c1_ns2_dep0_img_comp", "CVE-nonFixable_low-c1_ns2_dep0_img_comp",
+					"CVE-fixable_critical-c2_ns1_dep0_img_comp", "CVE-nonFixable_low-c2_ns1_dep0_img_comp",
+					"CVE-fixable_critical-c2_ns2_dep0_img_comp", "CVE-nonFixable_low-c2_ns2_dep0_img_comp",
+					"CVE-fixable_critical-w0_img_comp",
+					"CVE-nonFixable_low-w0_img_comp",
+					"CVE-fixable_critical-w1_img_comp",
+					"CVE-nonFixable_low-w1_img_comp",
+				},
+			},
+		},
+		// Test cases for CVE-specific search fields
+		{
+			name:  "View-based report filtering by CVE ID",
+			query: "CVE:CVE-fixable_critical-c1_ns1_dep0_img_comp",
+			expected: &viewBasedReportData{
+				deploymentNames: []string{"c1_ns1_dep0"},
+				imageNames:      []string{"c1_ns1_dep0_img"},
+				componentNames:  []string{"c1_ns1_dep0_img_comp"},
+				cveNames:        []string{"CVE-fixable_critical-c1_ns1_dep0_img_comp"},
+			},
+		},
+		{
+			name:  "View-based report filtering by CVSS score range",
+			query: "CVSS:>=7.0",
+			expected: &viewBasedReportData{
+				deploymentNames: []string{"c1_ns1_dep0", "c1_ns2_dep0", "c2_ns1_dep0", "c2_ns2_dep0"},
+				imageNames:      []string{"c1_ns1_dep0_img", "c1_ns2_dep0_img", "c2_ns1_dep0_img", "c2_ns2_dep0_img", "w0_img", "w1_img"},
+				componentNames:  []string{"c1_ns1_dep0_img_comp", "c1_ns2_dep0_img_comp", "c2_ns1_dep0_img_comp", "c2_ns2_dep0_img_comp", "w0_img_comp", "w1_img_comp"},
+				cveNames: []string{
+					"CVE-fixable_critical-c1_ns1_dep0_img_comp",
+					"CVE-fixable_critical-c1_ns2_dep0_img_comp",
+					"CVE-fixable_critical-c2_ns1_dep0_img_comp",
+					"CVE-fixable_critical-c2_ns2_dep0_img_comp",
+					"CVE-fixable_critical-w0_img_comp",
+					"CVE-fixable_critical-w1_img_comp",
+				},
+			},
+		},
+		//{
+		//	name:  "View-based report filtering by NVD CVSS score",
```

Comment:
Uncomment or remove?

Reply:
If this test isn't working, I think you should understand why.  It looks very similar to the test case above.  Impossible to really understand why without seeing the failures.

Reply:
@dashrews78  basically select framework does not return any result any results with this query string. it cannot be parsed I believe and I am not sure as to what is wrong in the string. Also I don't think these filters are exposed int he UI so customer wont use it. I have deleted these tests for now 
However I think POST endpoint should return an error if query string is not parsable

Reply:
The search string `NVD CVSS` looks valid, so the test shouldn't fail. You may want to check if the data is inserted correctly  and if the SQL query is generated correctly.

Reply:
In your test environment you can set `ROX_POSTGRES_QUERY_LOGGER=true` to see what the queries are.  But be warned.  It will spit out all the queries.  So you will want to put in debug statements in your test to know what test is running so you can find the query you are looking for.

Reply:
And I stand by my statement that we shouldn't just delete the test without a full understanding of why it didn't work.  Maybe the test is not valid.  But maybe it is.  Maybe it is exposing a flaw somewhere that we expose ourselves to if we don't at least understand why.

Reply:
@dashrews78  Both tests were incorrect
NVD CVSS test failed  because test is not inserting nvd score correctly. i have deleted  the test
<img width="1150" height="446" alt="Screenshot 2025-08-28 at 4 05 55 PM" src="https://github.com/user-attachments/assets/7332af30-b428-4ef6-9e9e-7813074a1d04" />

First occurence timestamp test failed because timestamp for CVE is actually earlier than 2020 which the test queries. I have changed query from > t to <t and test passes now.

Reply:
I have added it bck